packages/agent-base/src/index.ts

```diff
@@ -21,21 +21,37 @@ export type AgentConnectOpts = HttpConnectOpts | HttpsConnectOpts;
 const INTERNAL = Symbol('AgentBaseInternalState');
 
 interface InternalState {
+	tlsUpgradeOpts?: TlsUpgradeOpts;
 	defaultPort?: number;
 	protocol?: string;
 	currentSocket?: Duplex;
 }
 
+export interface TlsUpgradeOpts {
+	cert?: string;
+	key?: string;
+	ca?: string;
+}
+
+export interface BaseAgentOptions extends http.AgentOptions {
+	tlsUpgradeOpts?: TlsUpgradeOpts;
+}
+
 export abstract class Agent extends http.Agent {
 	private [INTERNAL]: InternalState;
 
 	// Set by `http.Agent` - missing from `@types/node`
 	options!: Partial<net.TcpNetConnectOpts & tls.ConnectionOptions>;
 	keepAlive!: boolean;
 
-	constructor(opts?: http.AgentOptions) {
+	constructor(opts?: BaseAgentOptions) {
+		const tlsUpgradeOpts = opts?.tlsUpgradeOpts;
+		delete opts?.tlsUpgradeOpts;
+
```

Comment:
maybe:

`const { tlsUpgradeOpts, ...restOpts } = opts;
super(restOps);
this[INTERNAL] = {
			tlsUpgradeOpts,
		};`

?